### PR TITLE
Configuring lint

### PR DESCRIPTION
**Pull request by KarishmaBothara**
To configure this in WebStorm, follow https://www.jetbrains.com/help/webstorm/prettier.html and do the same for tslint

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "printWidth": 120,
+  "trailingComma": "es5",
+  "tabWidth": 4,
```
Comment (benxgao) [inline .prettierrc]: 2

File: tslint.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
```
Comment (benxgao) [inline tslint.json]: Eslint supports TS now, maybe just focus on Eslint?

https://dev.to/robertcoopercode/using-eslint-and-prettier-in-a-typescript-project-53jb

https://javascriptplayground.com/typescript-eslint/

https://eslint.org/blog/2019/01/future-typescript-eslint

File: types/ExtrinsicSignature.ts [hunk 3]
```diff
@@ -2,17 +2,33 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import { Address, Balance, Call, EcdsaSignature, Ed25519Signature, ExtrinsicEra, Index, MultiSignature, Sr25519Signature } from '@polkadot/types/interfaces/runtime';
-import { IExtrinsicSignature, IKeyringPair } from '@polkadot/types/types';
+import {
+    Address,
```
Comment (benxgao) [inline types/ExtrinsicSignature.ts]: Saw a lot of unnecessary changes which is not related to linting and formatting? Is it performed by linting tools?

Comment (KarishmaBothara) [reply types/ExtrinsicSignature.ts]: yes...Its formatted with tslint.. alphabetical ordering of imports and multiple imports can be combined into one (no duplicate imports) as specified in tslint

File: types/Extrinsic.ts [hunk 4]
```diff
@@ -30,77 +30,83 @@ export interface ExtrinsicValueV4 {
...
-    } else if (isU8a(value)) {
-      // here we decode manually since we need to pull through the version information
-      const signature = new PlugExtrinsicSignatureV1(value, { isSigned });
-      const method = createType('Call', value.subarray(signature.encodedLength));
-
-      return {
-        method,
-        signature
-      };
+    /**
```
Comment (benxgao) [inline types/Extrinsic.ts]: Could we place these in a separated PR, and link a task ticket if possible, so that we all know the ticket background and I am able to give a proper review opinion? What do you think @holygits

File: package.json [hunk 5]
```diff
@@ -19,5 +33,13 @@
   "bugs": {
     "url": "https://github.com/plugblockchain/plug-api-types/issues"
   },
-  "homepage": "https://github.com/plugblockchain/plug-api-types#readme"
+  "homepage": "https://github.com/plugblockchain/plug-api-types#readme",
+  "husky": {
+    "hooks": {
+      "pre-commit": "pretty-quick --staged --pattern 'types/*' && npx tslint -c tslint.json --project ."
```
Comment (benxgao) [inline package.json]: Have you tried with writing some unformatted code, and see if the tools actually help with automatic formatting? Which could be evidence of task done?

Comment (KarishmaBothara) [reply package.json]: yes

**Review by jordy25519 — APPROVED** [review]

**Review by benxgao — APPROVED** [review]
